app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
     }
+    private title;
+    private form;
+    private mode_state;
+    private applicationId;
+    private actionId;
+    private jobTemplate;
+    private poolTemplate;
+    private pickedPool = new FormControl(null);
+    private error;
 
-    public NcjParameterExtendedType = NcjParameterExtendedType;
-
-    public jobNameParam: NcjParameterWrapper;
-
-    public applicationId: string;
-    public actionId: string;
-    public title = "";
-    public jobTemplate: NcjJobTemplate;
-    public poolTemplate;
-    public form: FormGroup;
-    public jobFormGroup: FormGroup;
-    public pickedPool = new FormControl(null);
-    public otherParameters: NcjParameterWrapper[];
-    public error: ServerError;
-
-    private _formValue: any;
-
-    // Constructor create data structure from reading json files
     constructor(
         public formBuilder: FormBuilder,
         private pythonRpcService: PythonRpcService,
         private route: ActivatedRoute,
         private router: Router,
-        private templateService: NcjTemplateService) { }
-
-    public ngOnInit() {
+        private templateService: NcjTemplateService) {
+             this.mode_state = "None";
+             this.form = new FormGroup({a: new FormControl()});
+    }
+    private ngOnInit() {
         this.route.params.subscribe((params) => {
             this.applicationId = params["applicationId"];
             this.actionId = params["actionId"];
-            this._updateTitle();
-            this._getTemplates();
-        });
-        this.route.queryParams.subscribe((params) => {
-            if (params.formParams) {
-                try {
-                    const value = JSON.parse(params.formParams);
-                    this._formValue = value;
-                    this.form.setValue(value);
-                } catch (e) {
-                    log.warn("Invalid form param. Not valid json.", params.formParams as any);
-                }
-            }
+            this.templateService.getTemplates(this.applicationId, this.actionId).subscribe((templates) => {
+                this.jobTemplate = templates.job;
+                this.poolTemplate = templates.pool;
+                this.title = `Run ${this.actionId} from ${this.applicationId}`;
+                this.createForms();
+            });
         });
     }
-
-    public createForms() {
-        let parameterKeys = Object.keys(this.jobTemplate.parameters);
+    private createForms() {
+        let jobParameters = [];
+        let poolParameters = [];
+        if (this.jobTemplate && this.jobTemplate.parameters){
+            jobParameters = Object.keys(this.jobTemplate.parameters);
+        }
+        if (this.poolTemplate && this.poolTemplate.parameters){
+            poolParameters = Object.keys(this.poolTemplate.parameters);
+        }
         let fg = {};
-        for (let key of parameterKeys) {
+
+        for (let key of jobParameters) {
             if ("defaultValue" in this.jobTemplate.parameters[key]) {
```

Comment:
this is actually would not be very useful because we do object.defaultValue after to access it.